tidy3d/components/geometry.py

```diff
@@ -2059,3 +2060,81 @@ def normalize(v):
 # geometries that can be used to define structures.
 GeometryFields = (Box, Sphere, Cylinder, PolySlab)
 GeometryType = Union[GeometryFields]
+
+
+class GeometryGroup(Geometry):
+    """A collection of Geometry objects that can be called as a single geometry object."""
+
+    geometries: List[GeometryType] = pydantic.Field(
+        [],
+        title="Geometries",
+        description="List of geometries in a single grouping. "
+        "Can provide significant performance enhancement in ``Structure`` when all geometries are "
+        "assigned the same medium.",
+        min_items=1,
+    )
+
+    @property
```

Comment:
Should we cache this?

Reply:
Geometry doesn't have a `_hash`, otherwise we could just cache all properties.  There was this issue with multiple inheritance that never got resolved.

Reply:
right now, only `Simulation` has a hash

Reply:
Every model has a `__hash__` thought. Are you just saying that `make_static` will not work?

Reply:
As in it will not make the `GeometryGroup` static. I think that could still be fine, the hash may be faster than getting all the geometry bounds?

Reply:
hm, we could give it a try. (move `cache` to `base.py` I guess?)